### PR TITLE
Armoured Shoes fixes and Code clean up

### PR DESCRIPTION
**Pull request by Nakalon**
### TLDR: Organized Feet.dm , fixed a couple of oversights regarding boots

While investigating this claim:

image

I discovered the feet.dm file was extremely messy.
There were a couple of duplicated vars, and some missing vars.
And oddly, I found out due to a possible oversight caused by this lack of orderliness Steel Plated boots could also be repaired by sewing.

In order to fix this I sorted all shoes that have an armour value into the same subtype and deleted any duplicated vars!

I also added crit protection to these armoured shoes and a hitsound, the same way these are done in other types of leather armour.

The investigation came to a close with one answer:
image

I guess this is not something to worry about regarding armour values!

## Why It's Good For The Game

Coming from a different codebase I found Ratwood codebase to be extremely messy.

This should improve readability, thus helping any coder that wishes to work with shoes! 

## Changelog

- Code: All armoured shoes are now a subtype of shoes/roguetown/armor
- Code: Cleaned up feet.dm by removing duplicated vars, commented out things that were never fully removed, and tidying up how the different type paths are sorted so that items that share the same subtype are bundled together.
- Added: Hitsound and Crit Prevention (Blunt and Twist) to all armoured shoes
- Fix: Steel Plated Boots are no longer repairable by sewing
- Code: Changed Steel Plated Boots type path from boots/armored to armor/steel for readability and to make way for more armored Boots that are not made out of steel

### CODE REVIEW

**Comment by Nakalon** [comment]
Those "mapping" changes are just a typepath change. I avoided doing things the lazy way.

**Comment by Red-Honk** [comment]
Seems like our PRs won't clash, and will complement one another, good.

Still I wonder why my PR has a red X while yours does not. https://github.com/Rotwood-Vale/Ratwood-Keep/pull/1132.

**Comment by john-pawson** [comment]
> Seems like our PRs won't clash, and will complement one another, good.
> 
> Still I wonder why my PR has a red X while yours does not. #1132.

Usually merge issues or problems with the code, try launching it to see if it throws you any problems, if that doesn't do it, then check to see if the branch is updated

**Comment by Nakalon** [comment]
> Seems like our PRs won't clash, and will complement one another, good.
> 
> Still I wonder why my PR has a red X while yours does not. #1132.

Left a comment there explaining why!

File: code/modules/clothing/rogueclothes/feet.dm [hunk 1]
```diff
@@ -78,70 +65,61 @@
...
-	name = "babouche"
-	desc = "Simple cheap shoes."
-	gender = PLURAL
-	icon_state = "shalal"
-	item_state = "shalal"
-	sewrepair = TRUE
-	armor = list("blunt" = 25, "slash" = 20, "stab" = 25, "bullet" = 0, "laser" = 0,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
 
-/obj/item/clothing/shoes/roguetown/boots/armor
+/obj/item/clothing/shoes/roguetown/boots //This subtype is for armoured shoes
```
Comment (Cyprex) [inline code/modules/clothing/rogueclothes/feet.dm:77]: How do you feel about /armoured or something instead?

Comment (Nakalon) [reply code/modules/clothing/rogueclothes/feet.dm:77]: This decision was made before I decided to replace a lot of subtypes, I was afraid to mess with a lot of files and have this in PR review hell. I will turn boots into armour or armor or armoured as you see fit!

**Comment by Red-Honk** [comment]
> > Seems like our PRs won't clash, and will complement one another, good.
> > Still I wonder why my PR has a red X while yours does not. #1132.
> 
> Left a comment there explaining why!

Could you like give me a comment on how to manually check what's wrong with the codes?; A mini tutorial would be nice, because now I wonder what's wrong with this one: https://github.com/Rotwood-Vale/Ratwood-Keep/pull/1033.

**Comment by Nakalon** [comment]
> > > Seems like our PRs won't clash, and will complement one another, good.
> > > Still I wonder why my PR has a red X while yours does not. #1132.
> > 
> > 
> > Left a comment there explaining why!
> 
> Could you like give me a comment on how to manually check what's wrong with the codes?; A mini tutorial would be nice, because now I wonder what's wrong with this one: #1033.

sure, press here

image

then here

image

press build

image

this will appear

image


Now you know whats wrong! If you need help reach out to me in discord, im June/Desdemona there, or Pinksuzuki

You should ALWAYS try to compile your code with ctrl+shift+b and pressing "build" at the very least.

You may also test it with F5.

Doing this will point you to the errors right away!
Hope this helps!

File: code/modules/clothing/rogueclothes/feet.dm (hunk 1 above)
Comment (Nakalon) [reply code/modules/clothing/rogueclothes/feet.dm]: Resolved! Changed from "boots" to "armor" in the same vein as other armors are done!

**Comment by katsukai** [comment]
Appreciate the work, good job.

**Comment by Cyprex** [comment]
Thanks a bunch for this!
